### PR TITLE
ci: Changes to CI configuration and support ruby 2.5 and 2.6

### PR DESCRIPTION
**Pull request by Mikemosca**
This Pr address removal of 2.3 and 2.4 and adding support for 2.6

2.7 will be handled under a separate PR because changes may be needed to core.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -398,8 +398,8 @@ If you have issues with the APIs or have a question about the Watson services, s
 
 Tested on:
 
-* MRI Ruby (RVM): 2.3.7, 2.4.4, 2.5.1
```
Comment (mamoonraja) [inline README.md:401]: Maybe we leave a warning for now that support for old versions will be deprecated in next major release.

File: ibm_watson.gemspec [hunk 2]
```diff
@@ -12,7 +12,7 @@ Gem::Specification.new do |spec|
   spec.summary = "Official client library to use the IBM Watson Services"
   spec.homepage = "https://www.github.com/watson-developer-cloud"
   spec.licenses = ["Apache-2.0"]
-  spec.required_ruby_version = ">= 2.3"
```
Comment (mamoonraja) [inline ibm_watson.gemspec:15]: In reference to my previous comment, maybe we leave this same and change it during next major release.

File: README.md (hunk 1 above)
Comment (Mikemosca) [reply README.md:401]: good call

File: ibm_watson.gemspec (hunk 2 above)
Comment (Mikemosca) [reply ibm_watson.gemspec]: also had to changes appveyor back to 2.3 for the build to pass.

**Pull request merged**